src/assets/scripts/client/game/GameController.js

```diff
@@ -581,7 +588,10 @@ class GameController {
     _onWindowFocus(event) {
         this.game.focused = true;
 
-        TimeKeeper.setPause(false);
+        // if was already manually paused when lost focus, respect that
+        if (!$('html').hasClass(SELECTORS.CLASSNAMES.PAUSED)) {
```

Comment:
~Is this actually needed?  All this method is trying to do is change the `focused` and `pause` values.  Pretty sure any UI feedback would be covered by your updates in `._onWindowBlur()`.  I'm also a little wary of calling `TimeKeeper` based on the state of an html element.~

I see what you're trying to do here.  We (well I am) are a big fan of exclusive `if` statements.  Meaning, when we don't have the data we need to continue, `return`.  So with that in mind, I would reverse the logic here to something like: 
```js
if ($('html').hasClass(SELECTORS.CLASSNAMES.PAUSED)) {
    return;
}
```

Last thing here, avoid using `$('html')` to select DOM elements.  Doing so means we have to start at the very top of the DOM tree to find what we're looking for.  Instead, use one of the cached elements.  Not sure if we have one in this class for the pause controls, but if we do we should use it.

Reply:
>  avoid using `$('html')` ...  Instead, use one of the cached elements.

Inline vars and `$()` appear to be used throughout this class, cached elements have been commented out where they were implemented. Something about tests breaking?
https://github.com/openscope/openscope/blob/3763f6e966350690699bea6f00a21147c0e3490b/src/assets/scripts/client/game/GameController.js#L78-L85

Reply:
Ideally, we would move all the button-related code to `UiController` which "owns" the buttons.

Reply:
done